daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -2419,6 +2422,25 @@ case class DFDLCheckConstraintsExpr(nameAsParsed: String, fnQName: RefQName,
 
 }
 
+case class DFDLCheckRangeInclusiveExpr(nameAsParsed: String, fnQName: RefQName, args: List[Expression])
+  extends FunctionCallBase(nameAsParsed, fnQName, args) {
+
+  override lazy val children = args
+
+  override lazy val compiledDPath = {
+    checkArgCount(3)
+    val compiledDPath = args(0).compiledDPath
+    val rangeFrom: Int = args(1).text.toInt
+    val rangeTo: Int = args(2).text.toInt
+    val res = new CompiledDPath(DFDLCheckRangeInclusive(compiledDPath, rangeFrom, rangeTo))
+    res
+  }
+  override def targetTypeForSubexpression(subexpr: Expression): NodeInfo.Kind = NodeInfo.AnyAtomic
```

Comment:
I think this is where you need to handle type information about the subexpressions. The targetType for the first paramater should probably be Numeric. The targetType for the min/max parameters should be the inherentType of the first parameter.

With this type information, Daffodil will insert the appropriate conversions (or error if not allowed) to convert the min/max parameters to the type of the node. This does mean the ``DFDLCheckRangeInclusive`` class will need to have logic to evaluate those subexpressions and handle different numeric types.